src/sdk/cpp/lib/src/schema.cpp

```diff
@@ -54,6 +54,12 @@ BooleanType::BooleanType()
 NumberType::NumberType()
     : ValueType{"number"} {}
 
+//
+// IntegerType
+//
+IntegerType::IntegerType()
```

Comment:
I assume these type definitions are related to json types. I'm not aware about an specific integer type in json (it is a part of number-type).  Please check if we mix up json and 'internal' types here.

Reply:
I think it's required here https://github.com/boschglobal/iot-event-analytics/blob/develop/resources/rules.schema.json#L70 but I could be wrong.

Reply:
We will keep it for now. There is an "integer" type available and used in our schemas, but we have to think about to change this and make it more robust / clear. (see https://json-schema.org/understanding-json-schema/reference/numeric.html#integer -- > e.g. switch to { "type": "number", "multipleOf": 1.0 }  to be on the safe side @halvfigur Thanks for this suggestion)